lib/manager/components/validation/MobilityDataValidationResult.js

```diff
@@ -0,0 +1,161 @@
+import { ListGroupItem, Table } from 'react-bootstrap'
+import Markdown from 'markdown-to-jsx'
+
+import toSentenceCase from '../../../common/util/to-sentence-case'
+import {
+  mobilityDataValidationErrorMapping,
+  validationErrorIconLookup
+} from '../../util/version'
+
+import rules from './rules.json'
+
+// from https://stackoverflow.com/a/4149671
+function unCamelCase (s) {
+  return s
+    .split(/(?=[A-Z])/)
+    .join(' ')
+    .toLowerCase()
+}
+
+const NoticeTable = ({ headerOverides = {
+  'stopSequence1': 'Stop seq-uence 1',
+  'stopSequence2': 'Stop seq-uence 2'
+}, notices }) => {
+  if (notices.length === 0) return null
+
+  const headers = Object.keys(notices[0])
+
+  return (
+    <Table bordered className='table-fixed' fill hover striped style={{display: 'relative', borderCollapse: 'collapse'}}>
+      <thead>
+        <tr>
+          {headers.map((header) => (
+            <th style={{position: 'sticky', top: 0, background: '#fefefe'}}>
+              {headerOverides[header] || toSentenceCase(unCamelCase(header))}
+            </th>
+          ))}
+        </tr>
+      </thead>
+      <tbody>
+        {notices.map((notice) => (
+          <tr>
+            {headers.map((header, index) => {
+              const FieldWrapper =
+                (header === 'fieldValue' || header === 'message') ? 'pre' : React.Fragment
+
+              let field = notice[header]
+              if (header.endsWith('Km') || header.endsWith('Kph')) {
+                field = Math.round(field)
+              }
+
+              return (
+                <td key={`${header}-${index}`}>
+                  <FieldWrapper>{field}</FieldWrapper>
+                </td>
+              )
+            })}
+          </tr>
+        ))}
+      </tbody>
+    </Table>
+  )
+}
+
+// eslint-disable-next-line complexity
+const renderNoticeDetail = (notice) => {
+  switch (notice.code) {
+    case 'too_many_rows':
+      notice.csvRowNumber = notice.rowNumber
+    case 'fare_transfer_rule_duration_limit_type_without_duration_limit':
+    case 'fare_transfer_rule_duration_limit_without_type':
+    case 'fare_transfer_rule_missing_transfer_count':
+    case 'fare_transfer_rule_with_forbidden_transfer_count':
+      notice.filename = 'fare_transfer_rules.txt'
+    case 'empty_file':
+    case 'emtpy_row':
+    case 'missing_timepoint_column':
+    case 'missing_required_file':
+    case 'missing_recommended_file':
+    case 'unknown_file':
+      return (
+        <ul>
+          {notice.sampleNotices.map((notice) => (
+            <li>
+              {notice.filename}
+              {notice.csvRowNumber && `: row ${notice.csvRowNumber}`}
+            </li>
+          ))}
+        </ul>
+      )
+    default:
+      return (
+        <NoticeTable notices={notice.sampleNotices} />
+      )
+  }
+}
+
+const MobilityDataValidationResult = (props) => {
+  const { notice } = props
+  const rule = rules.find((rd) => rd.rule === notice.code)
```

Comment:
would `indexOf` be more efficient here?

Reply:
It might be but I think it would be dirtier code wouldn't it